baseband/vdif/header.py

```diff
@@ -208,7 +208,14 @@ def fromvalues(cls, edv=False, **kwargs):
         # Some defaults that are not done by setting properties.
         kwargs.setdefault('legacy_mode', True if edv is False else False)
         kwargs['edv'] = edv
-        return super(VDIFHeader, cls).fromvalues(edv, **kwargs)
+        time = kwargs.pop('time', None)
+        sample_rate = kwargs.pop('sample_rate', None)
+        self = super(VDIFHeader, cls).fromvalues(edv, **kwargs)
+        if sample_rate is not None and 'sample_rate' in self._properties:
+            self.sample_rate = sample_rate
+        if time is not None:
+            self.set_time(time, sample_rate)
```

Comment:
I think `VDIFHeader.fromvalues(edv=0xab..., time=...)` will now fail, since its `set_time` doesn't accept a `sample_rate`. Easiest (and probably most sensible?) might be to pass on `sample_rate` there as well. It may be that you then do not need the workaround for `from_mark5b_header` any more.

Reply:
Or rather, it does not accept a second argument. Which in turn suggests that, just to be safe, we write this here as `sample_rate=sample_rate`.